### ISSUE TITLE
SH1107 display not working

### ISSUE THREAD

**Issue by joveaaron** [wokwi/wokwi-features#799]
**Describe the bug**
Code works, display doesn't. Serial console works fine, that's how I managed to tell if it was running. It was working just fine at home, idk what happened

**To Reproduce**
Load up my project, hit run. https://wokwi.com/projects/REDACTED

**Expected behavior**
Text is printed and screen is cleared.

**Environment (please complete the following information):**
 - OS: Ubuntu
 - Browser Chrome AND Firefox
 - Version (Firefox 114.0.1) (Chrome 103.0.5060.53)

**Additional context**
When loading the project I get these errors on both browsers.
```
Uncaught Error: Minified React error #418; visit https://reactjs.org/docs/error-decoder.html?invariant=418 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at lg (framework-128cfde67f3541f2.js:9:46339)
    at i (framework-128cfde67f3541f2.js:9:121283)
    at oD (framework-128cfde67f3541f2.js:9:99073)
    at framework-128cfde67f3541f2.js:9:98940
    at oO (framework-128cfde67f3541f2.js:9:98947)
    at ox (framework-128cfde67f3541f2.js:9:93986)
    at x (framework-128cfde67f3541f2.js:33:1364)
    at MessagePort.T (framework-128cfde67f3541f2.js:33:1894)
framework-128cfde67f3541f2.js:9 Uncaught Error: Minified React error #418; visit https://reactjs.org/docs/error-decoder.html?invariant=418 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at lg (framework-128cfde67f3541f2.js:9:46339)
    at i (framework-128cfde67f3541f2.js:9:121106)
    at oD (framework-128cfde67f3541f2.js:9:99073)
    at framework-128cfde67f3541f2.js:9:98940
    at oO (framework-128cfde67f3541f2.js:9:98947)
    at ox (framework-128cfde67f3541f2.js:9:93986)
    at x (framework-128cfde67f3541f2.js:33:1364)
    at MessagePort.T (framework-128cfde67f3541f2.js:33:1894)
2framework-128cfde67f3541f2.js:9 Uncaught Error: Minified React error #418; visit https://reactjs.org/docs/error-decoder.html?invariant=418 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at lg (framework-128cfde67f3541f2.js:9:46339)
    at i (framework-128cfde67f3541f2.js:9:121283)
    at oD (framework-128cfde67f3541f2.js:9:99073)
    at framework-128cfde67f3541f2.js:9:98940
    at oO (framework-128cfde67f3541f2.js:9:98947)
    at ox (framework-128cfde67f3541f2.js:9:93986)
    at x (framework-128cfde67f3541f2.js:33:1364)
    at MessagePort.T (framework-128cfde67f3541f2.js:33:1894)
6framework-128cfde67f3541f2.js:9 Uncaught Error: Minified React error #418; visit https://reactjs.org/docs/error-decoder.html?invariant=418 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at lg (framework-128cfde67f3541f2.js:9:46339)
    at i (framework-128cfde67f3541f2.js:9:121106)
    at oD (framework-128cfde67f3541f2.js:9:99073)
    at framework-128cfde67f3541f2.js:9:98940
    at oO (framework-128cfde67f3541f2.js:9:98947)
    at ox (framework-128cfde67f3541f2.js:9:93986)
    at x (framework-128cfde67f3541f2.js:33:1364)
    at MessagePort.T (framework-128cfde67f3541f2.js:33:1894)
framework-128cfde67f3541f2.js:9 Uncaught Error: Minified React error #423; visit https://reactjs.org/docs/error-decoder.html?invariant=423 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at i (framework-128cfde67f3541f2.js:9:120775)
    at oD (framework-128cfde67f3541f2.js:9:99073)
    at framework-128cfde67f3541f2.js:9:98940
    at oO (framework-128cfde67f3541f2.js:9:98947)
    at oE (framework-128cfde67f3541f2.js:9:95699)
    at ox (framework-128cfde67f3541f2.js:9:94254)
    at x (framework-128cfde67f3541f2.js:33:1364)
    at MessagePort.T (framework-128cfde67f3541f2.js:33:1894)
```

And when running project (only Firefox I believe):
```
Failed to paste diagram parts: SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data
    NextJS 26
        to
        m
        uU
        oV
        o
        x
        T
        60053
        p
        63840
        p
        64448
        p
        73935
        p
        20745
        p
        38060
        p
        1775
        p
        <anonymous>
        O
        <anonymous>
        u
        <anonymous>
```

**Comment by joveaaron** [comment]
Tested on Opera GX LVL 5 109.0.5097.108 and it still doesn't work.
Error log:
```
Uncaught Error: Minified React error #418; visit https://reactjs.org/docs/error-decoder.html?invariant=418 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at lg (framework-128cfde67f3541f2.js:9:46459)
    at i (framework-128cfde67f3541f2.js:9:121106)
    at oD (framework-128cfde67f3541f2.js:9:99073)
    at framework-128cfde67f3541f2.js:9:98940
    at oO (framework-128cfde67f3541f2.js:9:98947)
    at ox (framework-128cfde67f3541f2.js:9:93986)
    at x (framework-128cfde67f3541f2.js:33:1364)
    at MessagePort.T (framework-128cfde67f3541f2.js:33:1894)
framework-128cfde67f3541f2.js:9 Uncaught Error: Minified React error #423; visit https://reactjs.org/docs/error-decoder.html?invariant=423 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at i (framework-128cfde67f3541f2.js:9:120775)
    at oD (framework-128cfde67f3541f2.js:9:99073)
    at framework-128cfde67f3541f2.js:9:98940
    at oO (framework-128cfde67f3541f2.js:9:98947)
    at oE (framework-128cfde67f3541f2.js:9:95699)
    at ox (framework-128cfde67f3541f2.js:9:94254)
    at x (framework-128cfde67f3541f2.js:33:1364)
    at MessagePort.T (framework-128cfde67f3541f2.js:33:1894)
```
Please, look into this. it was working just fine yesterday. in fact, Opera GX caches websites when my browser is closed, so when I opened it when I came home, it hadn't changed at all and it's working! It's def a problem on your side, because the cached website works fine, and when I update the code and run it, it doesnt work again!!! @urish

**Comment by urish** [comment]
Hi, thanks for reporting. We've recently upgraded the ESP32 Arduino Core to the latest version, 3.0.1. It could be that the SH110X library is not compatible with the new Arduino core, or that the new Arduino Core has some glitches in the simulation. 

I was able to get it to work by reverting to an earlier Arduino Core version, by changing line 11 in diagram.json to 

```json
      "attrs": {"builder": "esp32-core-2.0.7"}
```

Result:

![image](https://github.com/wokwi/wokwi-features/assets/892318/d0f3369b-489e-4144-b4ad-08a7b210d281)
